### ISSUE TITLE
isin on joined data frames broke in 2021.5.0

### ISSUE THREAD

**Issue by jendrikjoe** [dask/dask#7765]
**What happened**:

A call to isin on a joined dataframe fails with `TypeError: only list-like objects are allowed to be passed to isin(), you passed a [str]`  in the distributed version.

**What you expected to happen**:

isin to execute as expected.

**Minimal Complete Verifiable Example**:

```python
from dask.distributed import Client

client = Client(memory_limit=0)

import dask.dataframe as dd
import pandas as pd

from nxml.helper.dask import extract_index_columns


df1 = pd.DataFrame(["a", "b", "c"], columns=["a"])
df2 = pd.DataFrame(["d", "e", "f"], columns=["1"])
df1 = dd.from_pandas(df1, npartitions=2)
df2 = dd.from_pandas(df2, npartitions=2)
ddf = df1.join(df2)

print(ddf["1"].isin(["a"]).compute())
```

**Anything else we need to know?**:

This appeared with the update from `2021.4.1`.

**Environment**:

- Dask version: 2021.5.1
- Python version: 3.9.4
- Operating System: macOS
- Install method (conda, pip, source): pip

**Comment by jrbourbeau** [comment]
Thanks for reporting @jendrikjoe! It looks like this has been fixed in the current `main` branch of `dask`:

```python
﻿﻿In [1]: from dask.distributed import Client
   ...:
   ...: client = Client(memory_limit=0)
   ...:
   ...: import dask.dataframe as dd
   ...: import pandas as pd
   ...:
   ...: #from nxml.helper.dask import extract_index_columns
   ...:
   ...:
   ...: df1 = pd.DataFrame(["a", "b", "c"], columns=["a"])
   ...: df2 = pd.DataFrame(["d", "e", "f"], columns=["1"])
   ...: df1 = dd.from_pandas(df1, npartitions=2)
   ...: df2 = dd.from_pandas(df2, npartitions=2)
   ...: ddf = df1.join(df2)
   ...:
   ...: print(ddf["1"].isin(["a"]).compute())
   ...:
0    False
1    False
2    False
Name: 1, dtype: bool
```

Could you retry with the development version of `dask` to confirm your issue has been resolved? (FWIW we're also releasing `dask=2021.06.0` today, so you could also try that release once it's out)

**Comment by jendrikjoe** [comment]
Sorry for the long delay in replying, `2021.06.0` actually resolves the issue 👍 Thank you very much 🎉
